napari/plugins/_builtins.py

```diff
@@ -29,3 +36,125 @@ def napari_get_reader(path: Union[str, List[str]]) -> ReaderFunction:
         function that returns layer_data to be handed to viewer._add_layer_data
     """
     return image_reader_to_layerdata_reader(magic_imread)
+
+
+@napari_hook_implementation(trylast=True)
+def napari_write_image(path: str, data: Any, meta: dict) -> bool:
+    """Our internal fallback image writer at the end of the plugin chain.
+
+    Parameters
+    ----------
+    path : str
+        Path to file, directory, or resource (like a URL).
+    data : array or list of array
+        Image data. Can be N dimensional. If the last dimension has length
+        3 or 4 can be interpreted as RGB or RGBA if rgb is `True`. If a
+        list and arrays are decreasing in shape then the data is from an image
+        pyramid.
+    meta : dict
+        Image metadata.
+
+    Returns
+    -------
+    bool : Return True if data is successfully written.
+    """
+    imsave(path, data)
+    return True
+
+
+@napari_hook_implementation(trylast=True)
+def napari_write_points(path: str, data: Any, meta: dict) -> bool:
+    """Our internal fallback points writer at the end of the plugin chain.
+
+    Append `.csv` extension to the filename if it is not already there.
```

Comment:
That's fine, but if no extension is provided then we should add one, i.e. `'file_name' -> 'file_name.csv'`. I can use `os.path.splitext(path)[1] == ''` to test if there is no extension, but the problem there is it will fail with paths like `"the.name.of.my.path.isnt.good"` which someone then wants a `csv` on. Curious what you recommend

Reply:
i've fixed this now, in that case i mention above, they can always add `.csv` at the end and then it's good